grapple/types/images.py

```diff
@@ -12,34 +8,29 @@
 )
 
 from ..registry import registry
-from ..utils import resolve_queryset
+from ..utils import resolve_queryset, get_media_item_url
 from .structures import QuerySetList
 
 
-def get_image_url(cls):
-    url = ""
-    if hasattr(cls, "url"):
-        url = cls.url
-    else:
-        url = cls.file.url
-
-    if url[0] == "/":
-        return settings.BASE_URL + url
-    return url
-
-
 class BaseImageObjectType(graphene.ObjectType):
     width = graphene.Int(required=True)
     height = graphene.Int(required=True)
     src = graphene.String(required=True)
+    url = graphene.String(required=True)
     aspect_ratio = graphene.Float(required=True)
     sizes = graphene.String(required=True)
 
+    def resolve_url(self, info):
+        """
+        Get the uploaded image url.
+        """
+        return get_media_item_url(self)
+
     def resolve_src(self, info):
         """
-        Get url of the original uploaded image.
+        Deprecated. Use the `url` attribute.
```

Comment:
This deprecation can be added to the GraphQL schema through the `deprecation_reason` scalar argument.
```python
src = graphene.String(required=True, deprecation_reason="Use the `url` attribute")
```

Reply:
I knew I did not search well enought. Kudos for the tip!